validator/test/fixtures/resource_value_pattern/constraints/data.yaml

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+list:
+  - apiVersion: constraints.gatekeeper.sh/v1alpha1
+    kind: GCPResourceValuePatternConstraintV1
+    metadata:
+      name: required_billing_id_on_project
+    spec:
+      severity: high
+      match:
+        gcp:
+          target: ["organization/*"]
+          exclude: [] # optional, default is no exclusions
+      parameters:
+        mode: allowlist
+        asset_types:
+          - "cloudresourcemanager.googleapis.com/Project"
+        field_name: "labels.billing_id"
+        pattern: "^\\d{4}$"
+        required: true
+
+  - apiVersion: constraints.gatekeeper.sh/v1alpha1
+    kind: GCPResourceValuePatternConstraintV1
+    metadata:
+      name: optional_billing_id_on_multiple_assets
+    spec:
+      severity: high
+      match:
+        gcp:
+          target: ["organization/*"]
+          exclude: [] # optional, default is no exclusions
+      parameters:
+        mode: allowlist
+        asset_types:
+          - "cloudresourcemanager.googleapis.com/Project"
+          - "compute.googleapis.com/Instance"
+        field_name: "labels.billing_id"
+        required: false
+        pattern: "^\\d{4}$"
```

Comment:
Is it possible that users might want to filter for and against something?  Rather than "mode" and "pattern", would "allow" and "deny" be more flexible?  Is it possible they might want to have a list of allowable patterns?

Reply:
That's an interesting idea. @AdrienWalkowiak as the OG of this PR, what are your thoughts?